File: app/model/XMLContest.java
package model;

/**
 * Created by gnufede on 12/05/14.
 */

import java.util.Date;
import com.fasterxml.jackson.annotation.JsonView;
import org.apache.commons.lang3.builder.HashCodeBuilder;
import org.bson.types.ObjectId;

public class XMLContest {
    public String name;
    public Date startDate;
    public String xml;

    @Override
    public int hashCode() {
        return new HashCodeBuilder(7, 61).append(startDate).append(xml).toHashCode();
    }

}

File: app/model/XMLContest.scala
package model

/**
 * Created by gnufede on 12/05/14.
 */

import org.joda.time.DateTime
import com.fasterxml.jackson.annotation.JsonView
import org.apache.commons.lang3.builder.HashCodeBuilder
import org.bson.types.ObjectId

class XMLContest (var name: String, var startDate: DateTime, var headers: Map[String, String], var xml: String){
  override def hashCode: Int = {
    return new HashCodeBuilder(7, 61).append(startDate).append(xml).toHashCode
  }

}